Sources/Parser.swift

```diff
@@ -111,11 +127,15 @@ public class TokenParser {
   }
 
   public func compileFilter(_ token: String) throws -> Resolvable {
```

Comment:
This method isn't part of the review, but this is the same comment for these 3 functions: why do these need to be `public`?

Reply:
While in custom filters the same methods can be accessed through `context.environment`, in custom nodes they will be accessed through `TokenParser` which `environment` property is prived, so these methods are private to allow custom nodes implementations to use them.